### PR TITLE
[native] Fix E2E test framework for CDF functions

### LINKED ISSUE #20715: [Native] E2E tests for probability functions cannot be run
## Expected Behaviour
E2E tests for probability functions are contained in the file [AbstractTestNativeProbabilityFunctionQueries.java](https://github.com/prestodb/presto/blob/master/presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeProbabilityFunctionQueries.java). Since this is an abstract class, a separate subclass implementation is required to run these tests.

## Current Behaviour
There are currently no subclasses of `AbstractTestNativeProbabilityFunctionQueries`. Therefore, none of the tests inside the file can be run.

## Possible Solution
Create a subclass of `AbstractTestNativeProbabilityFunctionQueries` and implement the abstract methods.

## Steps to Reproduce
This can be confirmed with a quick [github code search](https://github.com/search?q=repo%3Aprestodb%2Fpresto%20AbstractTestNativeProbabilityFunctionQueries&type=code). The class is referenced only once, there are no subclasses.

### PR DESCRIPTION
**Pull request by ashkrisk**
Resolves #20715

Added a new test class, `TestNativeProbabilityFunctionQueries`, which extends the abstract class `AbstractTestNativeProbabilityFunctionQueries`. Previously, it was not possible to run the E2E tests contained in the abstract class, since it had no subclasses.

The E2E test for `binomial_cdf` is explicitly disabled since it is failing. It can be re-enabled once the issue is resolved.

```
== NO RELEASE NOTE ==
```

### CODE REVIEW

File: presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeProbabilityFunctionQueries.java [hunk 1]
```diff
@@ -23,10 +23,23 @@
 public abstract class AbstractTestNativeProbabilityFunctionQueries
         extends AbstractTestQueryFramework
 {
+    // The superclass method returns an ExpectedQueryRunner,
+    // but this class expects a more specific subclass - QueryRunner.
+    // Make this requirement explicit
+    @Override
```
Comment (aditi-pandit) [inline presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeProbabilityFunctionQueries.java:29]: Why have you added this ?

Other classes like AbstractTestNativeAggregations.java don't have these functions and simply state 
QueryRunner queryRunner = (QueryRunner) getExpectedQueryRunner(); in the createTables  function.

Comment (ashkrisk) [reply presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeProbabilityFunctionQueries.java:29]: This abstract class has some expectations of its subclasses:
- `createExpectedQueryRunner` must be overridden
- `createExpectedQueryRunner` must return an object of type `QueryRunner`.

Unless these requirements are satisfied, the following line will throw a Runtime Exception:
`QueryRunner queryRunner = (QueryRunner) getExpectedQueryRunner()`

This abstract function formalises the above two requirements. If a subclass fails to satisfy the requirement, there will be a compile-time error.

In short, adding this takes advantage of Java's type checking and turns potential runtime errors into compile-time errors, which will be noticed and fixed earlier in the development cycle.

Comment (aditi-pandit) [reply presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeProbabilityFunctionQueries.java:29]: Ah I see. So the AbstractTestQueryFramework was written for presto-java tests that compare PrestoJava (has QueryRunner) with H2QueryRunner (has ExpectedQueryRunner). In presto-native tests, presto_cpp (has QueryRunner) is compared with PrestoJava(has QueryRunner). As the presto-java side is used to create tables (since that support is limited in C++) so `QueryRunner queryRunner = (QueryRunner) getExpectedQueryRunner()` works implicitly. You are making this explicit. 

While this is a good idea, it might be better to do it at a more generic level in presto-native-execution than specifically for this class. I would derive AbstractNativeTestQueryFramework from AbstractTestQueryFramework and change the implementation.

But lets not complicate this here I feel. We can follow the same style as other native tests for now. If there is strong feeling to change ExpectedQueryRunner to QueryRunner for the java query runner we can do it with a generic solution across all tests.

Comment (ashkrisk) [reply presto-native-execution/src/test/java/com/facebook/presto/nativeworker/AbstractTestNativeProbabilityFunctionQueries.java]: That's fair. I've removed the method overrides, it should now match the other test classes.

**Comment by aditi-pandit** [comment]
@ashkrisk : Please can you merge your commits.

**Comment by ashkrisk** [comment]
@aditi-pandit squashed the commits
